### PR TITLE
feat: joint id loader

### PR DESCRIPTION
**Pull request by HLasse**
Adds a `split` column to the regional split dataframe for easy integration with other loaders. Added a function for loading either the original splits or the regional splits.

### CODE REVIEW

File: psycop/common/feature_generation/application_modules/save_dataset_to_disk.py [hunk 1]
```diff
@@ -76,7 +79,7 @@ def filter_by_split_ids(
 
 def get_split_id_df(split_name: SplitName) -> pd.DataFrame:
     """Get a dataframe with the splits ids."""
-    split_id_df = load_ids(
+    split_id_df = load_original_ids(
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/application_modules/save_dataset_to_disk.py:82]: `load_patient_level_split_ids`?

File: psycop/common/feature_generation/loaders/raw/load_ids.py [hunk 2]
```diff
@@ -3,25 +3,30 @@
...
 
 
-def load_ids(split: SplitName, n_rows: int | None = None) -> pd.DataFrame:
-    """Loads ids for a given split.
+def load_original_ids(split: SplitName, n_rows: int | None = None) -> pd.DataFrame:
+    """Loads ids for a given split based on the original data split.
 
     Args:
-        split (str): Which split to load IDs from. Takes either "train", "test" or "val". # noqa: DAR102
+        split: Which split to load IDs from. Takes either "train", "test" or "val".
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/loaders/raw/load_ids.py:29]: [goforit](https://app.graphite.dev/macros/Aarhus-Psychiatry-Research)

File: psycop/common/feature_generation/loaders/raw/load_ids.py [hunk 3]
```diff
@@ -34,3 +39,46 @@ def load_ids(split: SplitName, n_rows: int | None = None) -> pd.DataFrame:
...
+        n_rows: Number of rows to return. Defaults to None.
+
+    Returns:
+        pd.DataFrame: Only dw_ek_borger column with ids
+    """
+    if split_type == "original":
+        return load_original_ids(split=split_name, n_rows=n_rows)
+    if split_type == "geographical":
+        return load_region_ids(split=split_name, n_rows=n_rows)
+    raise ValueError(f"split_type {split_type} is not allowed")
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/loaders/raw/load_ids.py:84]: What's the advantage to having two separate functions?

Comment (MartinBernstorff) [reply psycop/common/feature_generation/loaders/raw/load_ids.py:84]: If we are keeping this, it looks like an excellent case for a match statement 👍 

(Then we can ignore the value-error)

File: psycop/common/feature_generation/loaders/raw/load_text.py [hunk 4]
```diff
@@ -149,6 +149,7 @@ def load_text_sfis(
 def load_text_split(
     text_sfi_names: str | Iterable[str],
     split_name: Sequence[SplitName],
+    split_type: Literal["original", "geographical"] = "original",
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/loaders/raw/load_text.py:152]: Ah, this is probably why you want to wrap the two functions. Instead, perhaps the split here should be a function? (Dependency-injected).

**Review by MartinBernstorff — COMMENTED** [review]
Looks good Lasse, really glad you did this! A few comments, up for a chat as well if you'd like 👍

File: psycop/common/feature_generation/loaders/raw/load_ids.py (hunk 2 above)
Comment (MartinBernstorff) [reply psycop/common/feature_generation/loaders/raw/load_ids.py:29]: Graphite ❤️

File: psycop/common/feature_generation/application_modules/save_dataset_to_disk.py (hunk 1 above)
Comment (HLasse) [reply psycop/common/feature_generation/application_modules/save_dataset_to_disk.py:82]: stratified_by_outcome_splits

**Pull request merged**